experiment/coverage_utils.py

```diff
@@ -0,0 +1,348 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utility functions for coverage report generation."""
+
+import os
+import multiprocessing
+import json
+import queue
+
+from common import experiment_utils as exp_utils
+from common import new_process
+from common import benchmark_utils
+from common import fuzzer_utils
+from common import logs
+from common import filestore_utils
+from common import experiment_path as exp_path
+from common import filesystem
+from database import utils as db_utils
+from database import models
+from experiment.build import build_utils
+from experiment import reporter
+
+logger = logs.Logger('coverage_utils')  # pylint: disable=invalid-name
+
+COV_DIFF_QUEUE_GET_TIMEOUT = 1
+
+
+def update_coverage_to_bucket():
+    """Copies the coverage reports to gcs bucket."""
+    report_dir = reporter.get_reports_dir()
+    src_dir = get_coverage_report_dir()
+    dst_dir = exp_path.filestore(report_dir)
+    filestore_utils.cp(src_dir, dst_dir, recursive=True, parallel=True)
+
+
+def generate_cov_reports(experiment_config: dict):
```

Comment:
Use full names
generate_coverage_reports
generate_coverage_report
for functions